cpp/src/aztec/honk/sumcheck/relations/relation.test.cpp

```diff
@@ -21,33 +32,31 @@ template <class FF> class SumcheckRelation : public testing::Test {
-        // Note: z_perm and z_perm_shift can be any linear poly for the sake of the tests but should not be be equal to
-        // each other In order to avoid a trivial computation in the case of the grand_product_computation_relation.
-        // Values here were chosen so that output univariates in tests are small positive numbers.
-        auto z_perm_shift = Univariate<6>({ 0, 1, 2, 3, 4, 5 }); // this is not real shifted data
-        auto q_m = Univariate<6>({ 1, 2, 3, 4, 5, 6 });
-        auto q_l = Univariate<6>({ 1, 2, 3, 4, 5, 6 });
-        auto q_r = Univariate<6>({ 1, 2, 3, 4, 5, 6 });
-        auto q_o = Univariate<6>({ 1, 2, 3, 4, 5, 6 });
-        auto q_c = Univariate<6>({ 1, 2, 3, 4, 5, 6 });
-        auto sigma_1 = Univariate<6>({ 1, 2, 3, 4, 5, 6 });
-        auto sigma_2 = Univariate<6>({ 1, 2, 3, 4, 5, 6 });
-        auto sigma_3 = Univariate<6>({ 1, 2, 3, 4, 5, 6 });
-        auto id_1 = Univariate<6>({ 1, 2, 3, 4, 5, 6 });
-        auto id_2 = Univariate<6>({ 1, 2, 3, 4, 5, 6 });
-        auto id_3 = Univariate<6>({ 1, 2, 3, 4, 5, 6 });
-        auto lagrange_first = Univariate<6>({ 1, 2, 3, 4, 5, 6 });
-        auto lagrange_last = Univariate<6>({ 1, 2, 3, 4, 5, 6 });
-        auto pow_zeta = Univariate<6>({ 1, 1, 1, 1, 1, 1 });
+        BarycentricData<FF, univariate_length, extension_length> barycentric_2_to_max =
+            BarycentricData<FF, univariate_length, extension_length>();
+        auto extended_univariate = barycentric_2_to_max.extend(input_univariate);
+        auto w_l = Univariate<extension_length>(extended_univariate);
+        auto w_r = Univariate<extension_length>(extended_univariate);
+        auto w_o = Univariate<extension_length>(extended_univariate);
+        auto z_perm = Univariate<extension_length>(extended_univariate);
+        auto z_perm_shift = Univariate<extension_length>(extended_univariate); // this is not real shifted data
+        auto q_m = Univariate<extension_length>(extended_univariate);
+        auto q_l = Univariate<extension_length>(extended_univariate);
+        auto q_r = Univariate<extension_length>(extended_univariate);
+        auto q_o = Univariate<extension_length>(extended_univariate);
+        auto q_c = Univariate<extension_length>(extended_univariate);
+        auto sigma_1 = Univariate<extension_length>(extended_univariate);
+        auto sigma_2 = Univariate<extension_length>(extended_univariate);
+        auto sigma_3 = Univariate<extension_length>(extended_univariate);
+        auto id_1 = Univariate<extension_length>(extended_univariate);
+        auto id_2 = Univariate<extension_length>(extended_univariate);
+        auto id_3 = Univariate<extension_length>(extended_univariate);
+        auto lagrange_first = Univariate<extension_length>(extended_univariate);
+        auto lagrange_last = Univariate<extension_length>(extended_univariate);
+        auto pow_zeta = Univariate<extension_length>(extended_univariate);
```

Comment:
Since we're allocating all the `Univariate`s to be the same, we don't actually need to name them. You could instead just initialize the array as follows:

```suggestion
        auto extended_univariate = barycentric_2_to_max.extend(input_univariate);
        std::array<Univariate<extension_length>, bonk::StandardArithmetization::NUM_POLYNOMIALS>
        for (size_t i = 0; i < bonk::StandardArithmetization::NUM_POLYNOMIALS; ++i) {
            extended_edges[i] = extended_univariate;
        }
        return extended_univariate;
```

Tiny note: the return type of `barycentric_2_to_max.extend()` is already `Univariate<extension_length>`, so the conversion here wasn't necessary ;)

Reply:
Done. I am not using different inputs as this is done in the `sumcheck_round.test.cpp`.